content/2022-02-28-v0_4-release.md

```diff
@@ -0,0 +1,92 @@
+
+### 💌 Validation & Authorization
+
+Making use of Rust’s strong type system, the `CoreGroup` layer underneath the `MlsGroup` API not only ensures that every incoming MLS message is syntactically and semantically validated according to the MLS spec, but also that messages created using the `MlsGroup` API fulfill these requirements. Before any messages are finally applied to a group, the API exposes the message to the application along with all information required to make its own policy decisions.
+
+All validation steps are [indexed and documented](https://openmls.tech/book/message_validation.html), as well as thoroughly tested.
+
+## 🏗️ Still Not Finished: MLS Keeps Improving
+
+For a while now, MLS has been “close to a working group last call”, but as implementations have matured and academic research on the protocol has progressed, issues and improvement proposals have been steadily trickling in. As a result, at the time of release of OpenMLS v0.4, there is still no working group last call and no additional draft of the specification has been released. Consequently, v0.4 is [not quite feature-complete](https://github.com/openmls/openmls/issues?q=is%3Aissue+is%3Aopen+label%3A%22mls-spec+change%22), but advances the implementation to draft 12+.
+
+## 🧑🏼‍🔧 More Improvements Under the Hood
+
+To achieve all of the milestones listed above, OpenMLS has seen many improvements under the hood.
+
+- 🌳 TreeSync: The tree implementation supporting OpenMLS has been re-written in a style that allows a diff-merge process and that exposes a clear abstraction boundary between the ratchet tree and the rest of the implementation.
+- ⚠️ Errors: OpenMLS now uses the `thiserror` crate throughout. Instead of module-wide error types, every function now returns an error enum that contains only the errors that can actually occur during the course of the function.
+- 😱 Almost panic-free: Although not quite there yet, we have made a lot of progress in transitioning to a panic-free implementation, such that even in the event of an implementation bug we can ensure that OpenMLS functions will not panic.
+- ⏱️ Parallelization: The performance-critical functions of OpenMLS around the encryption of key material to other group members is now parallelized using the `rayon` crate.
+
+## 💼 Future Work
+
+The MLS specification is still in motion and there are many things that need to be improved before we can move towards the next OpenMLS release. If you have a suggestion, please let us know by joining our [Zulip instance](https://openmls.zulipchat.com) or by filing an issue in our [GitHub repo](https://github.com/openmls/openmls). Or send us an email at [hello@openmls.tech](mailto:hello@openmls.tech).
+
+## 💯 Metrics
+
+Here are some data points that give an overview of our activity since the release of the roadmap in October 2021.
+
+- [Closed issues since October 2021: 1](https://github.com/openmls/openmls/issues?q=is%3Aissue+is%3Aclosed+updated%3A2021-10-01..2022-02-28+)88
```

Comment:
These look a little funny. Let's putt the entire number in the links (same below).